rust/src/encodings/plain.rs

```diff
@@ -310,6 +307,23 @@ impl<'a> PlainDecoder<'a> {
     }
 }
 
+fn make_chunked_requests(
+    indices: &[u32],
+    byte_width: usize,
+    block_size: usize,
+) -> Vec<Range<usize>> {
+    let mut chunked_ranges = vec![];
+    let mut start: usize = 0;
+    for (i, w) in indices.windows(2).enumerate() {
```

Comment:
so this is comparing sequential pairs of the indices but originally it's comparing each index with `start`?

Reply:
This is the same logic, but just make it more `rust-style`